common/test/acceptance/tests/lms/test_progress_page.py

```diff
@@ -0,0 +1,277 @@
+            self.assertEqual(self._get_section_score(), (1, 2))
+
+    @contextmanager
+    def _logged_in_session(self, staff=False):
+        """
+        Ensure that the user is logged in and out appropriately at the beginning
+        and end of the current test.
+        """
+        self.logout_page.visit()
+        try:
+            if staff:
+                auto_auth(self.browser, "STAFF_TESTER", "staff101@example.com", True, self.course_id)
+            else:
+                auto_auth(self.browser, self.USERNAME, self.EMAIL, False, self.course_id)
+            yield
+        finally:
+            self.logout_page.visit()
+
+
+class ProgressPageTest(ProgressPageBaseTest):
+    """
+    Test that the progress page reports scores from completed assessments.
+    """
+    def test_progress_page_shows_scored_problems(self):
+        self._check_progress_page_with_scored_problem()
+
+
+@ddt.ddt
+class PersistentGradesTest(ProgressPageBaseTest):
+    """
+    Test that grades for completed assessments are persisted
+    when various edits are made.
+    """
+    def setUp(self):
+        super(PersistentGradesTest, self).setUp()
+        self.instructor_dashboard_page = InstructorDashboardPage(self.browser, self.course_id)
+
+    def _change_subsection_structure(self):
+        """
+        Adds a unit to the subsection, which
+        should not affect a persisted subsection grade.
+        """
+        with self._logged_in_session(staff=True):
+            self.course_outline.visit()
+            subsection = self.course_outline.section(self.SECTION_NAME).subsection(self.SUBSECTION_NAME)
+            subsection.expand_subsection()
+            subsection.add_unit()
+
+    def _set_staff_lock_on_subsection(self, locked):
+        """
+        Sets staff lock for a subsection, which should hide the
+        subsection score from students on the progress page.
+        """
+        with self._logged_in_session(staff=True):
+            self.course_outline.visit()
+            subsection = self.course_outline.section_at(0).subsection_at(0)
+            subsection.set_staff_lock(locked)
+            self.assertEqual(subsection.has_staff_lock_warning, locked)
+
+    def _change_weight_for_problem(self):
+        """
+        Changes the weight of the problem, which should not affect
+        persisted grades.
+        """
+        with self._logged_in_session(staff=True):
+            self.course_outline.visit()
+            self.course_outline.section_at(0).subsection_at(0).expand_subsection()
+            unit = self.course_outline.section_at(0).subsection_at(0).unit(self.UNIT_NAME).go_to()
+            component = unit.xblocks[1]
+            component.edit()
+            component_editor = ComponentEditorView(self.browser, component.locator)
+            component_editor.set_field_value_and_save('Problem Weight', 5)
+
+    def _rescore_for_all(self):
+        """
+        Triggers a rescore for all students. Currently unimplemented.
```

Comment:
Why merge if this is unimplemented?  Is there a TODO ticket for implementing this?
Also, rescoring a problem DOES update the grade.  So the test is invalid for this case, right?